app/models/content_item_factory.rb

```diff
@@ -1,6 +1,6 @@
 class ContentItemFactory
-  def self.build(content_hash)
-    content_item_class(content_hash["document_type"]).new(content_hash)
+  def self.build(content_store_response)
```

Comment:
I don't think this needs to be renamed. What's being passed in was one of three potential things anyway.

Reply:
No, I think all three things are GdsApi::Responses. The only time it was actually getting a hash was in our tests, where we were giving it a hash.

The three possible things are:

- `request.env[:content_item]`
- `request_content_item(content_item_slug)`
- `request_content_item("/#{params[:slug]}")`

`request_content_item` always returns a GdsApi::Response, and this is also what's cached on `request.env[:content_item]` [here](https://github.com/alphagov/frontend/blob/main/lib/format_routing_constraint.rb#L17)

Reply:
I mean it's a passed in parameter and doesn't need to be renamed as part of this change. It could be renamed independently of everything else and nothing else will be affected.

Reply:
True, but the name is wrong - it's not a hash, it's a response.

This change is about clearing up the response / hash confusion, so I think it's legitimate to fix the parameter name here.

Reply:
Happy to pull it out into a separate commit if that would make it more palatable?

Reply:
All of the naming in general is confusing. But fine, this isn't a blocker. Separate commit would be better. There is so much going on.